channel-analysis/lowfee-routing-diversity/lowfee_routing_diversity.py

```diff
@@ -186,27 +210,35 @@ def get_lowfee_reachable_node_maxflows(proposed_new_peer=None, max_hops=None):
         min_channels = int(sys.argv[4])
     if len(sys.argv) >= 6:
         min_capacity = int(sys.argv[5])
-
+json_data = json.load(sys.stdin)
+json_data_root = getInputType(json_data)
+if inputType == InputType.UKN:
+    sys.stderr.write("Neither Channels (C-Lightning) nor Edges (LND) nodes found in input JSON ")
+    sys.exit(1)
+else:
+    print("Found %s input JSON" % inputType.value)
 i = 1
 num_inactive_channels = 0
-for chan in json.load(sys.stdin)["channels"]:
-    src_id = chan["source"]
+for chan in json_data_root:
+    if inputType == InputType.LND and (chan["node1_policy"] == None or chan["node2_policy"] == None):
+        continue
+    src_id = chan["source"] if inputType == InputType.CLI else chan["node1_pub"]
     if src_id not in id_to_node:
         node_to_id[i] = src_id
         id_to_node[src_id] = i
         nodes.add(i)
         i += 1
     src = id_to_node[src_id]
 
-    dest_id = chan["destination"]
+    dest_id = chan["destination"] if inputType == InputType.CLI else chan["node2_pub"]
     if dest_id not in id_to_node:
         node_to_id[i] = dest_id
         id_to_node[dest_id] = i
         nodes.add(i)
         i += 1
     dest = id_to_node[dest_id]
 
-    if not chan["active"]:
+    if (inputType == InputType.CLI and not chan["active"]) or (inputType == InputType.CLI and (chan["node1_policy"]["disabled"] or chan["node2_policy"]["disabled"])):
```

Comment:
should be `... (inputType != InputType.CLI and (chan["node1_policy"]["disabled"] ...`